File: python/program/Pelanggan.py
from Orang import Orang

# subclass Pelanggan
class Pelanggan(Orang):

    def __init__(self, noHp: str, email: str, alamat: str,
                 nama: str, usia: int, tanggalLahir: str,
                 kodePelanggan: str, saldo: str, kuantitas: int,
                 statusMember: str, riwayatTransaksi: str):
        # panggil constructor parent (Orang)
        super().__init__(noHp, email, alamat, nama, usia, tanggalLahir)
        # atribut khusus Pelanggan
        self._kodePelanggan = str(kodePelanggan)
        self._saldo = str(saldo)  # sekarang string
        self._kuantitas = int(kuantitas)
        self._statusMember = str(statusMember)
        self._riwayatTransaksi = str(riwayatTransaksi)

    # Getter & Setter Kode Pelanggan
    def getKodePelanggan(self) -> str:
        return self._kodePelanggan

    def setKodePelanggan(self, kodePelanggan: str) -> None:
        self._kodePelanggan = str(kodePelanggan)

    # Getter & Setter Saldo (string)
    def getSaldo(self) -> str:
        return self._saldo

    def setSaldo(self, saldo: str) -> None:
        self._saldo = str(saldo)

    # Getter & Setter Kuantitas
    def getKuantitas(self) -> int:
        return self._kuantitas

    def setKuantitas(self, kuantitas: int) -> None:
        self._kuantitas = int(kuantitas)

    # Getter & Setter Status Member
    def getStatusMember(self) -> str:
        return self._statusMember

    def setStatusMember(self, statusMember: str) -> None:
        self._statusMember = str(statusMember)

    # Getter & Setter Riwayat Transaksi
    def getRiwayatTransaksi(self) -> str:
        return self._riwayatTransaksi

    def setRiwayatTransaksi(self, riwayatTransaksi: str) -> None:
        self._riwayatTransaksi = str(riwayatTransaksi)

File: java/program/Pelanggan.java
public class Pelanggan extends Orang {
    // atribut tambahan khusus pelanggan
    private String kodePelanggan;     
    private String saldo;            
    private int kuantitas;           
    private String statusMember; // status keanggotaan (misalnya Gold, Silver)
    private String riwayatTransaksi; 

    // constructor untuk inisialisasi data pelanggan
    public Pelanggan(String noHp, String email, String alamat,
                     String nama, int usia, String tanggalLahir,
                     String kodePelanggan, String saldo, int kuantitas,
                     String statusMember, String riwayatTransaksi) {
        // panggil constructor parent (Orang) untuk isi data umum
        super(noHp, email, alamat, nama, usia, tanggalLahir);
        // isi data khusus pelanggan
        this.kodePelanggan = kodePelanggan;
        this.saldo = saldo;
        this.kuantitas = kuantitas;
        this.statusMember = statusMember;
        this.riwayatTransaksi = riwayatTransaksi;
    }

    // getter & setter untuk akses dan ubah data
    public String getKodePelanggan() {
        return kodePelanggan;
    }

    public void setKodePelanggan(String kodePelanggan) {
        this.kodePelanggan = kodePelanggan;
    }

    public String getSaldo() {
        return saldo;
    }

    public void setSaldo(String saldo) {
        this.saldo = saldo;
    }

    public int getKuantitas() {
        return kuantitas;
    }

    public void setKuantitas(int kuantitas) {
        this.kuantitas = kuantitas;
    }

    public String getStatusMember() {
        return statusMember;
    }

    public void setStatusMember(String statusMember) {
        this.statusMember = statusMember;
    }

    public String getRiwayatTransaksi() {
        return riwayatTransaksi;
    }

    public void setRiwayatTransaksi(String riwayatTransaksi) {
        this.riwayatTransaksi = riwayatTransaksi;
    }
}
